// js-info/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.runtime = exports.os = exports.isMobile = exports.javascriptEngine = exports.getInfo = exports.environment = exports.engine = exports.browserEngine = exports.browser = undefined;

var _memoize = require('lodash/memoize');

var _memoize2 = _interopRequireDefault(_memoize);

var _browser = require('./browser');

var _browser2 = _interopRequireDefault(_browser);

var _browserEngine = require('./browserEngine');

var _browserEngine2 = _interopRequireDefault(_browserEngine);

var _environment = require('./environment');

var _environment2 = _interopRequireDefault(_environment);

var _javascriptEngine = require('./javascriptEngine');

var _javascriptEngine2 = _interopRequireDefault(_javascriptEngine);

var _os = require('./os');

var _os2 = _interopRequireDefault(_os);

var _runtime = require('./runtime');

var _runtime2 = _interopRequireDefault(_runtime);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var engine = _javascriptEngine2.default;
var memoized = {
  isMobile: null
};

function isMobile() {
  if (!memoized.isMobile) memoized.isMobile = (0, _memoize2.default)(require('./isMobile').default);
  return memoized.isMobile();
}

function getInfo() {
  return {
    browser: _browser2.default.info,
    browserEngine: _browserEngine2.default.info,
    engine: engine.info,
    environment: _environment2.default.info,
    javascriptEngine: _javascriptEngine2.default.info,
    mobile: isMobile(),
    os: _os2.default.info,
    runtime: _runtime2.default.info
  };
}

exports.browser = _browser2.default;
exports.browserEngine = _browserEngine2.default;
exports.engine = engine;
exports.environment = _environment2.default;
exports.getInfo = getInfo;
exports.javascriptEngine = _javascriptEngine2.default;
exports.isMobile = isMobile;
exports.os = _os2.default;
exports.runtime = _runtime2.default;
exports.default = {
  get info() {
    return getInfo();
  },
  get mobile() {
    return isMobile();
  },
  browser: _browser2.default,
  browserEngine: _browserEngine2.default,
  engine,
  environment: _environment2.default,
  getInfo,
  javascriptEngine: _javascriptEngine2.default,
  isMobile,
  os: _os2.default,
  runtime: _runtime2.default
};
