// js-info/lib/javascriptEngine.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _memoize = require('lodash/memoize');

var _memoize2 = _interopRequireDefault(_memoize);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var getJavascriptEngine = null;

var JavascriptEngine = function () {
  function JavascriptEngine() {
    _classCallCheck(this, JavascriptEngine);
  }

  _createClass(JavascriptEngine, [{
    key: 'chakra',
    get: function get() {
      return this.value === 'chakra';
    }
  }, {
    key: 'info',
    get: function get() {
      return {
        chakra: this.chakra,
        jsc: this.jsc,
        nitro: this.nitro,
        spidermonkey: this.spidermonkey,
        v8: this.v8,
        value: this.value
      };
    }
  }, {
    key: 'jsc',
    get: function get() {
      return this.value === 'jsc' || this.nitro;
    }
  }, {
    key: 'nitro',
    get: function get() {
      return this.value === 'nitro';
    }
  }, {
    key: 'spidermonkey',
    get: function get() {
      return this.value === 'spidermonkey';
    }
  }, {
    key: 'v8',
    get: function get() {
      return this.value === 'v8';
    }
  }, {
    key: 'value',
    get: function get() {
      if (!getJavascriptEngine) {
        getJavascriptEngine = (0, _memoize2.default)(require('./getJavascriptEngine').default);
      }
      return getJavascriptEngine();
    }
  }]);

  return JavascriptEngine;
}();

exports.default = new JavascriptEngine();
