// js-info/lib/getBrowser/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = getBrowser;

var _ = require('..');

function isOpera() {
  if (!!window.opr && !!window.opr.addons) return true;
  if (!!window.opera) return true;
  var userAgent = window.navigator.userAgent;

  if (/\sopr/i.test(userAgent)) return true;
  return false;
} // https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser/9851769

function isFirefox() {
  if (typeof window.InstallTrigger !== 'undefined') return true;
  var userAgent = window.navigator.userAgent;

  if (_.os.ios && /fxios/i.test(userAgent)) return true;
  return false;
}

function isSafari() {
  if (/constructor/i.test(window.HTMLElement)) return true;
  if (window.safari && typeof window.safari !== 'undefined') {
    var pushNotification = window.safari.pushNotification;

    return pushNotification.toString() === '[object SafariRemoteNotification]';
  }
  var userAgent = window.navigator.userAgent;

  if (/safari/i.test(userAgent)) return true;
  return false;
}

function isIE() {
  /* eslint-disable spaced-comment */
  // prettier-ignore
  return (/*@cc_on!@*/false || !!window.document.documentMode
  );
  /* eslint-enable */
}

function isEdge() {
  if (!isIE() && !!window.StyleMedia) return true;
  var userAgent = window.navigator.userAgent;

  if (_.os.android && /\sedg/i.test(userAgent)) return true;
  return false;
}

function isChrome() {
  if (!!window.chrome && !!window.chrome.webstore) return true;
  var userAgent = window.navigator.userAgent;

  if (/chrome/i.test(userAgent)) return true;
  if (_.os.ios && /crios/i.test(userAgent)) return true;
  return false;
}

function getBrowser() {
  if (!_.runtime.browser) return 'na';
  if (isIE()) return 'ie';
  if (isEdge()) return 'edge';
  if (isFirefox()) return 'firefox';
  if (isOpera()) return 'opera';
  if (isChrome()) return 'chrome';
  if (isSafari()) return 'safari';
  return 'unknown';
}
